src/zopfli/zlib.py

```diff
@@ -1,14 +1,245 @@
+        res = bytearray(data[0])
+        self.bit = data[1]
+
+        if final:
+            self.lastbyte = b''
+            return res
+        else:
+            self.lastbyte = res[-32:]
+            return res[:-32]
+
+    def compress(self, string):
+        global MASTER_BLOCK_SIZE
+        self.buf.extend(bytearray(string))
+        if len(self.buf) > MASTER_BLOCK_SIZE:
+            out = bytearray()
+            if not self.raw and self.first:
+                out.extend(self._header())
+                self.first = False
+            out.extend(self._compress())
+            return bytes(out)
+        else:
+            return b''
+
+    def flush(self, mode=Z_FINISH):
+        def encodedalign(prev=None):
+            res = bytearray()
+            if prev:
+                z = bytearray(prev)
+            else:
+                z = bytearray()
+            z.extend([0, 0, 0])
+            tgtlen = 8 if len(z) <= 8 else 16
+            addlen = tgtlen - len(z)
+            z.extend((0,) * addlen)
+            res.append(bitseq2int(z[:8]))
+            if tgtlen == 16:
+                res.append(bitseq2int(z[8:]))
+            lbits = int2bitseq(0, 16)
+            res.append(bitseq2int(lbits[:8]))
+            res.append(bitseq2int(lbits[8:]))
+            nlbits = int2bitseq(65535, 16)
+            res.append(bitseq2int(nlbits[:8]))
+            res.append(bitseq2int(nlbits[8:]))
+            return res
+
+        if self.closed:
+            raise error
+        out = bytearray()
+        self.closed = mode == Z_FINISH
+        if not self.raw and self.first:
+            out.extend(self._header())
+            self.first = False
+        if mode == Z_NO_FLUSH:
+            return bytes(out)
+        out.extend(self._compress(mode == Z_FINISH))
+        if mode != Z_FINISH:
+            self.bit = self.bit % 8
+            # add void fixed block to align data to bytes
+            if self.bit:
+                work = int2bitseq(self.lastbyte.pop(), 8)[:self.bit]
+            else:
+                work = []
+            self.lastbyte.extend(encodedalign(work))
+            out.extend(self.lastbyte)
+            self.lastbyte = b''
+            self.bit = 0
+            if mode == Z_FULL_FLUSH:
+                self.prehist = bytearray()
+
+        if not self.raw and mode == Z_FINISH:
+            out.extend(self._tail())
+        return bytes(out)
+
+
+def compress(data, level=-1, **kwargs):
     """zlib.compress(data, **kwargs)
-    
-    """ + zopfli.__COMPRESSOR_DOCSTRING__  + """
+
+    """ + zopfli.__COMPRESSOR_DOCSTRING__ + """
     Returns:
       String containing a zlib container
     """
+    if 'numiterations' not in kwargs:
+        if level not in levit:
+            raise error
+        kwargs['numiterations'] = levit[level]
+
     kwargs['gzip_mode'] = 0
-    return zopfli.zopfli.compress(data, **kwargs)
+    return zopfli.zopfli.compress(bytes(data), **kwargs)
```

Comment:
Change this for some unittest.
Maybe change to 'y' in C code will be better solution, but this depends on compatibility with 2.7